business_requirement_deliverable/models/project.py

```diff
@@ -15,3 +15,10 @@ class Project(models.Model):
         Deliverables linked to this project.
         Currency of the Deliverables will be the one from this pricelist.'''
     )
+
+    @api.multi
+    def get_closest_ancestor_pricelist(self):
+        for project in self:
+            pricelist = self.env['product.pricelist'].browse(
+                project.pricelist_id.id)
```

Comment:
not sure this is enough.
Rule should be for the pricelist:
If pricelist in master project then use project.pricelist
else if pricelist is project parent then use parent_project.pricelist
else use partner.pricelist